r-package/grf/R/print.R

```diff
@@ -94,30 +94,47 @@ print.boosted_regression_forest <- function(x, ...) {
 #' @importFrom stats aggregate quantile
 #' @export
 print.tuning_output <- function(x, tuning.quantiles = seq(0, 1, 0.2), ...) {
-  grid <- x$grid
-  out <- lapply(colnames(grid)[-1], function(name) {
-    q <- quantile(grid[, name], probs = tuning.quantiles)
-    # Cannot form for example quintiles for mtry if the number of variables is
-    # less than 5, so here we just truncate the groups.
-    if (length(unique(q) < length(q))) {
-      q <- unique(q)
-    }
-    rank <- cut(grid[, name], q, include.lowest = TRUE)
-    out <- aggregate(grid[, "error"], by = list(rank), FUN = mean)
-    colnames(out) <- c(name, "error")
-    out
-  })
-
-  err <- x$error
-  params <- x$params[colnames(grid)[-1]]
-  opt <- formatC(c(err, params))
 
-  cat("Optimal tuning parameters: \n")
-  cat(paste0(names(opt), ": ", opt, "\n"))
-
-  cat("Average error by ", length(tuning.quantiles) - 1, "-quantile:\n", sep = "")
-  for (i in out) {
+  if (x$status == "failure") {
+    cat("Tuning status: failure.\n")
+    cat("This indicates tuning was attempted but failed, and we fell back to default parameters: \n\n")
+    params <- x$params
+    cat(paste0(names(params), ": ", params, "\n"))
+  } else if (x$status == "default") {
+    cat("Tuning status: default.\n")
+    cat("This indicates tuning was not attempted, and we used default parameters: \n\n")
```

Comment:
But tuning was attempted..? Will only print if tune.parameters = TRUE and tuning was attempted but params.default was best

Tuning means finding the best parameters, if the best was the default, then that's still a tuning result? Is some code bloat arising from all the different categories (in an already large tuning module) but see you must have a branch here in the current form because otherwise x$grid etc is empty...

Reply:
Ah, good point. I'll fix that immediately.

>  Is some code bloat arising from all the different categories (in an already large tuning module) but see you must have a branch here in the current form because otherwise x$grid etc is empty...

Sorry, I don't understand what you mean.